docs/proposals/multipart_trasnfer_mode.md

```diff
@@ -0,0 +1,296 @@
+            },
+            "pos": 0,
+            "size": 2500000,
+            "expires_in": 86400
+          },
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=1",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 2500000,
+            "size": 2500000,
+            "expires_in": 86400
+          },
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=2",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 5000000,
+            "size": 2500000,
+            "expires_in": 86400
+          },
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=3",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 7500000,
+            "expires_in": 86400
+          }
+        ],
+        "verify": {
+          "href": "https://lfs.mycompany.com/myorg/myrepo/multipart/verify",
+          "authenticated": true,
+          "header": {
+            "Authorization": "Basic 123abc123abc123abc123abc123="
+          },
+          "expires_in": 86400,
+          "params": {
+            "uploadId": "20492a4d0d84",
+            "partIds": [0, 1, 2, 3]
+          }
+        },
+        "abort": {
+          "href": "https://foo.cloud.com/storage/upload/20492a4d0d84",
+          "authenticated": true,
+          "header": {
+            "Authorization": "Basic 123abc123abc123abc123abc123="
+          },
+          "method": "DELETE",
+          "expires_in": 86400
+        }
+      }
+    }
+  ]
+}
+```
+
+## Uploaded Part Digest
+Some storage backends will support, or even require, uploading clients to send a digest of the uploaded part as part
+of the request. This is a useful capability even if not required, as it allows backends to validate each part
+separately as it is uploaded.
+
+To support this, `parts` request objects may include a `want_digest` value, which may be any value specified by
+[RFC-3230](https://tools.ietf.org/html/rfc3230) or [RFC-5843](https://tools.ietf.org/html/rfc5843) (the design for this
+feature is highly inspired by these RFCs).
+
+Possible values include a comma-separated list of q-factor flagged algorithms, such as SHA-256 and SHA-512.
+Of one or more of these are specified, the digest of the payload is to be specified by the client as part of
+the Digest header, using the format specified by [RFC-3230 section 4.3.2](https://tools.ietf.org/html/rfc3230#section-4.3.1).
+
+It should be noted that digests based on `MD5` or `SHA-1` algorithms are not supported by this transfer mode,
+as they are no longer to be considered secure. Therefore, the `contentMD5`, `MD5` and `SHA` values are not to be used as
+part of `want_digest` headers.
```

Comment:
Let's also specify that we only allow cryptographically secure hash algorithms in this field.  For example, one can specify `adler32` or `unixcksum` in a Digest header, but we're not going to implement that, either, because as a practical matter nobody uses those.